src/System.Private.CoreLib/src/System/Diagnostics/Eventing/EventPipePayloadDecoder.cs

```diff
@@ -29,20 +29,8 @@ internal static object[] DecodePayload(ref EventSource.EventMetadata metadata, R
                 Type parameterType = parameters[i].ParameterType;
                 if (parameterType == typeof(IntPtr))
                 {
-                    if (IntPtr.Size == 8)
-                    {
-                        // Payload is automatically updated to point to the next piece of data.
-                        decodedFields[i] = (IntPtr)ReadUnalignedUInt64(ref payload);
-                    }
-                    else if (IntPtr.Size == 4)
-                    {
-                        decodedFields[i] = (IntPtr)MemoryMarshal.Read<Int32>(payload);
-                        payload = payload.Slice(IntPtr.Size);
-                    }
-                    else
-                    {
-                        Debug.Assert(false, "Unsupported pointer size.");
-                    }
+                    decodedFields[i] = MemoryMarshal.Read<IntPtr>(payload);
+                    payload = payload.Slice(IntPtr.Size);
```

Comment:
It's been a little while since I wrote this code, but I don't believe that this is safe on arm.  At the time, MemoryMarshal.Read didn't handle the case where the read might be unaligned.  Has that changed?

Reply:
`MemoryMarshal.Read` always handled misaligned reads. You must be confusing this with something else.

Reply:
It is possible - in that case, carry on.